lib/gitlib.sh

```diff
@@ -40,6 +40,7 @@ BRANCH_REGEX="master|release-([0-9]{1,})\.([0-9]{1,})(\.([0-9]{1,}))*$"
 declare -A VER_REGEX=([release]="v(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(-alpha|-beta)*\.*(0|[1-9][0-9]*)?"
                       [dotzero]="v(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.0$"
                       [build]="([0-9]{1,})\+([0-9a-f]{5,40})"
+                      [cibuild]="([0-9]{1,})\+*([0-9a-f]{5,40})*"
```

Comment:
I'm not saying we should change `VER_REGEX[build]`. what I'm saying is that the bug is when the git branch was just tagged with a release string, it will only match `VER_REGEX[release]` with no additional suffix.

For example,
```console
$ git checkout v1.4.5
HEAD is now at 5a0a696... Kubernetes version v1.4.5
( source hack/lib/version.sh && KUBE_ROOT=$(pwd) kube::version::get_version_vars && kube::version::save_version_vars /dev/stdout )
KUBE_GIT_COMMIT='5a0a696437ad35c133c0c8493f7e9d22b0f9b81b'
KUBE_GIT_TREE_STATE='clean'
KUBE_GIT_VERSION='v1.4.5'
KUBE_GIT_MAJOR='1'
KUBE_GIT_MINOR='4'
```

I'm pretty sure `v1.4.5` will not match `${VER_REGEX[release]}\.${VER_REGEX[cibuild]}`.